From 94adacf7ebd602e8dd6f901279a3b164c12e4498 Mon Sep 17 00:00:00 2001
From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Tue, 21 Jul 2020 03:25:23 +0530
Subject: [PATCH 58/58] Bring back trichrome_chrome_apk

* Since M84, chromium trichrome_chrome_bundle doesn't
  play nice with bundletool and thus can't be signed.
* bundletool's aapt2 invocation fails with
  "duplication configuration in resource table.",
  see the bug for more details

Partial Revert of "Android: Remove trichrome_chrome_apk"

This reverts commit be712e648cb870eb125ce6832fec55859e3f56ec.

Bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1106115
---
 chrome/android/BUILD.gn | 46 ++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 45 insertions(+), 1 deletion(-)

diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 0304b73b1be2..0aa55e164a50 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -1566,6 +1566,10 @@ if (current_toolchain == default_toolchain) {
     is_monochrome = true
     is_bundle_module = true
   }
+  resource_packaging("trichrome_chrome_apk_pak_assets") {
+    is_monochrome = false
+    is_trichrome = true
+  }
   resource_packaging("trichrome_chrome_bundle_module_pak_assets") {
     is_monochrome = false
     is_trichrome = true
@@ -1966,7 +1970,7 @@ trichrome_library_apk_tmpl("trichrome_library_apk") {
 
   if (trichrome_synchronized_proguard) {
     webview_target = "//android_webview:trichrome_webview_apk"
-    chrome_target = ":trichrome_chrome_bundle"
+    chrome_target = ":trichrome_chrome_apk"
   }
 
   verify_manifest = _enable_manifest_verification
@@ -2006,6 +2010,46 @@ android_resources("trichrome_dummy_resources") {
   sources = [ "trichrome/res_dummy/values/strings.xml" ]
 }
 
+monochrome_public_apk_or_module_tmpl("trichrome_chrome_apk") {
+  version_code = trichrome_version_code
+  version_name = chrome_version_name
+  apk_name = "TrichromeChrome"
+  target_type = "android_apk"
+  use_trichrome_library = true
+  static_library_provider = ":trichrome_library_apk"
+  if (!is_java_debug) {
+    static_library_synchronized_proguard = trichrome_synchronized_proguard
+    if (trichrome_synchronized_proguard) {
+      resource_ids_provider_dep = "//android_webview:trichrome_webview_apk"
+    }
+  }
+  if (android_64bit_target_cpu) {
+    is_64_bit_browser = false
+    include_64_bit_webview = true
+  }
+}
+
+if (android_64bit_target_cpu) {
+  monochrome_public_apk_or_module_tmpl("trichrome_chrome_64_32_apk") {
+    version_code = trichrome_version_code
+    version_name = chrome_version_name
+    apk_name = "TrichromeChrome6432"
+    target_type = "android_apk"
+    use_trichrome_library = true
+    static_library_provider = ":trichrome_library_apk"
+    if (!is_java_debug) {
+      static_library_synchronized_proguard = trichrome_synchronized_proguard
+      if (trichrome_synchronized_proguard) {
+        resource_ids_provider_dep = "//android_webview:trichrome_webview_apk"
+      }
+    }
+    if (android_64bit_target_cpu) {
+      is_64_bit_browser = true
+      include_32_bit_webview = true
+    }
+  }
+}
+
 chrome_public_test_apk_manifest =
     "$root_gen_dir/chrome_public_test_apk_manifest/AndroidManifest.xml"
 chrome_public_test_vr_apk_manifest =
-- 
2.27.0

